server/sonar-server/src/main/java/org/sonar/server/computation/formula/SumCounter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.formula;
+
+import com.google.common.base.Optional;
+import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.measure.Measure;
+
+public class SumCounter implements Counter<SumCounter> {
+
+  private final String metricKey;
+
+  private Integer value;
+
+  public SumCounter(String metricKey) {
+    this.metricKey = metricKey;
+  }
+
+  @Override
+  public void aggregate(SumCounter counter) {
+    if (counter.getValue().isPresent()) {
+      addValueIfPresent(counter.getValue().get());
+    }
+  }
+
+  @Override
+  public void aggregate(Component component, CounterContext counterContext) {
+    Optional<Measure> measureOptional = counterContext.getMeasure(metricKey);
+    if (measureOptional.isPresent()) {
+      addValueIfPresent(measureOptional.get().getIntValue());
+    }
+  }
+
+  private void addValueIfPresent(Integer newValue) {
+    value = value == null ? newValue : value + newValue;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)